src/etc/licenseck.py

```diff
@@ -57,18 +57,18 @@
 
 def check_license(name, contents):
     # Whitelist check
-    for exception in exceptions:
-        if name.endswith(exception):
-            return True
+    if any(name.endswith(e) for e in exceptions):
+        return True
 
     # Xfail check
     firstlineish = contents[:100]
-    if firstlineish.find("ignore-license") != -1:
+    if "ignore-license" in firstlineish:
         return True
 
     # License check
     boilerplate = contents[:500]
-    if (boilerplate.find(license1) == -1 or boilerplate.find(license2) == -1) and \
-       (boilerplate.find(license3) == -1 or boilerplate.find(license4) == -1):
+    if (license1 not in boilerplate or license2 not in boilerplate) and \
```

Comment:
Yes, this change makes a lot of sense. I'll add it as a separate commit. It took me a while to figure out why I couldn't get it working on Python 2 and 3; it turns out Python 3 doesn't like backslashes in raw (`r`) strings, so it's easier to just escape backslashes manually.

Thank you for the code review, @jatinn!